### PR TITLE
Users can create BetOptions for Bet

### PR DESCRIPTION
**Pull request by ghost**
Pivotal: https://www.pivotaltracker.com/story/show/143167657
Added the ability for users to add a BetOption while viewing a particular Bet.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
   resources :bets do
     resources :user_bets
+    resources :bet_options
```
Comment (qr8r) [inline config/routes.rb]: You might want to read about shallow nesting for resources

File: app/views/bet_options/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<div class="container">
+  <hr>
+  <% if current_user %>
```
Comment (qr8r) [inline app/views/bet_options/new.html.erb]: This should perhaps be a `before_action` in the controller

File: app/views/bet_options/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <%= f.text_field :option_text, class: 'form-control' %>
+      </div>
+
+      <div class="form-group">
+        <%= f.submit 'Create bet option', class: 'btn btn-primary' %>
+      </div>
+    <% end %>
+
+  <% else %>
+    <h2>You are not logged in </h2>
```
Comment (qr8r) [inline app/views/bet_options/new.html.erb]: This should be a shared partial somewhere

**Review by qr8r — APPROVED** [review]
Minor comments

**Review by qr8r — CHANGES_REQUESTED** [review]
Woops, fix changes first please

File: config/routes.rb [hunk 4]
```diff
@@ -10,6 +10,7 @@
 
   resources :bets do
     resources :user_bets, shallow: true
+    resources :bet_options, shallow: true
```
Comment (qr8r) [inline config/routes.rb]: Does this PR implement all 7 crud actions?

File: spec/controllers/bet_options_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    subject { get :new, bet_id: bet.id }
+    let(:user) do
+      User.create(username: 'Bob', email: 'bob@mail.com', password: '123')
+    end
+    let(:bet) do
+      Bet.create(bet: 'Bet text',
+                 expires_at: DateTime.tomorrow,
+                 creator_id: user.id)
+    end
+    before { log_in(user) }
```
Comment (qr8r) [inline spec/controllers/bet_options_controller_spec.rb]: You should also test the case where the user is not logged in

**Review by qr8r — CHANGES_REQUESTED** [review]
I am able to add bets with no option text. Is that desired behaviour?

**Comment by ghost** [comment]
Yes, I haven't implemented any validation yet, I think it's a separate story on Pivotal.

**Comment by qr8r** [comment]
I dont believe it is currently a story. You can add it as a story, or fix it here. Ill approve both scenarios. Just link me the story if thats the route you choose to go

**Review by qr8r — APPROVED** [review]

**Comment by YarekTenko** [comment]
I'll resolve the conflicts once #36 is merged into master.

**Comment by jacquesporveau** [comment]
Looks like you have some merge conflicts. Also noticing this PR was opened about a week ago but there are commits in here that are 28+ days old. Are you sure you are trying to merge into the correct branch?

**Comment by ghost** [comment]
@jacquesporveau Thanks, fixed.

**Pull request merged**